### PR TITLE
WebOfScience rake tasks - the saga begins

### PR DESCRIPTION
**Pull request by darrenleeweber**
Connected to #253 

The beginning of `wos:{x}` rake tasks.

### Motivation
- This adds WOS API utils module/singleton, available in rails console
- Added `wos:publication` task, based on `sw:publication` task
- Added `wos:links` task to get all the link fields for a publication

### CODE REVIEW

**Comment by darrenleeweber** [comment]
### Example

```
$ bundle exec rake wos:console

From: /data/src/dlss/cap/sul_pub/lib/tasks/wos.rake @ line 27 :

    22:   end
    23: 
    24:   desc 'Drop into a console with WOS clients'
    25:   task :console => :environment do
    26:     require 'pry'
 => 27:     binding.pry
    28:   end
    29: 
    30:   # TODO: could this use a CSV string of ids?
    31:   desc 'Retrieve and print a single publication by WosItemId'
    32:   task :publication, [:wos_id] => :environment do |_t, args|

[1] pry(main)> wos_client
=> #<WebOfScience::Client:0x0055744d985c58 @auth_code="U3RhbmZvcmRVX1NXOjJAIyRTdFVuaQ", @log_level=:info>
[2] pry(main)> wos_queries
=> #<WebOfScience::Queries:0x0055744d91a9d0
 @database="WOK",
 @wos_client=#<WebOfScience::Client:0x0055744d985c58 @auth_code="U3RhbmZvcmRVX1NXOjJAIyRTdFVuaQ", @log_level=:info>>
[3] pry(main)> Author.count
=> 111795
```

**Comment by darrenleeweber** [comment]
### Example wos:publication

```
$  bundle exec rake wos:publication['MEDLINE:24452614']
I, [2017-11-01T22:05:34.195351 #32445]  INFO -- : SOAP request: 
I, [2017-11-01T22:05:34.387104 #32445]  INFO -- : SOAP response (status 200)
I, [2017-11-01T22:05:34.814615 #32445]  INFO -- : SOAP request: 
I, [2017-11-01T22:05:35.833099 #32445]  INFO -- : SOAP response (status 200)
<?xml version='1.0'?>
<REC r_id_disclaimer='ResearcherID data provided by Clarivate Analytics'>
  <UID>MEDLINE:24452614</UID>
  <static_data>
    <summary>
      <EWUID>
        <WUID coll_id='MEDLINE'/>
        <edition value='MEDLINE.MEDLINE'/>
      </EWUID>
      <pub_info coverdate='2014-Jan-22' edate='2014-01-22' has_abstract='Y' medium='Print' model='Electronic' pubday='22' pubmonth='Jan' pubtype='Journal' pubyear='2014' sortdate='2014-01-22' vol='3'>
        <page begin='e93'>e93</page>
      </pub_info>
      <titles count='4'>
        <title type='item'>
          Identifying druggable targets by protein microenvironments matching:
          application to transcription factors.
        </title>
        <title type='source'>CPT: pharmacometrics &amp; systems pharmacology</title>
        <title type='abbrev_iso'>CPT Pharmacometrics Syst Pharmacol</title>
        <title type='source_abbrev'>CPT Pharmacometrics Syst Pharmacol</title>
      </titles>
      <names count='2'>
        <name display='Y' role='author' seq_no='1'>
          <display_name>Liu, T</display_name>
          <full_name>Liu, T</full_name>
          <initials>T</initials>
        </name>
        <name display='Y' role='author' seq_no='2'>
          <display_name>Altman, R B</display_name>
          <full_name>Altman, R B</full_name>
          <initials>RB</initials>
        </name>
      </names>
      <doctypes count='1'>
        <doctype>Journal Article</doctype>
      </doctypes>
    </summary>
    <fullrecord_metadata>
      <languages count='1'>
        <language type='primary'>English</language>
      </languages>
      <normalized_languages count='1'>
        <language type='primary'>English</language>
      </normalized_languages>
      <normalized_doctypes count='1'>
        <doctype>Article</doctype>
      </normalized_doctypes>
      <addresses count='1'>
        <address_name>
          <address_spec addr_no='1'>
            <full_address>Department of Genetics, Stanford University, Stanford, California, USA.</full_address>
          </address_spec>
        </address_name>
      </addresses>
      <fund_ack>
        <grants complete='Y' count='3'>
          <grant>
            <grant_agency>NIGMS NIH HHS</grant_agency>
            <grant_ids count='1'>
              <grant_id>R01 GM102365</grant_id>
            </grant_ids>
            <country>United States</country>
            <acronym>GM</acronym>
          </grant>
          <grant>
            <grant_agency>NLM NIH HHS</grant_agency>
            <grant_ids count='1'>
              <grant_id>R01 LM005652</grant_id>
            </grant_ids>
            <country>United States</country>
            <acronym>LM</acronym>
          </grant>
          <grant>
            <grant_agency>NIGMS NIH HHS</grant_agency>
            <grant_ids count='1'>
              <grant_id>U54 GM072970</grant_id>
            </grant_ids>
            <country>United States</country>
            <acronym>GM</acronym>
          </grant>
        </grants>
      </fund_ack>
      <abstracts count='1'>
        <abstract>
          <abstract_text>
            <p>
              Druggability of a protein is its potential to be modulated by
              drug-like molecules. It is important in the target selection
              phase. We hypothesize that: (i) known drug-binding sites contain
              advantageous physicochemical properties for drug binding, or
              "druggable microenvironments" and (ii) given a target, the
              presence of multiple druggable microenvironments similar to those
              seen previously is associated with a high likelihood of
              druggability. We developed DrugFEATURE to quantify druggability by
              assessing the microenvironments in potential small-molecule
              binding sites. We benchmarked DrugFEATURE using two data sets. One
              data set measures druggability using NMR-based screening.
              DrugFEATURE correlates well with this metric. The second data set
              is based on historical drug discovery outcomes. Using the
              DrugFEATURE cutoffs derived from the first, we accurately
              discriminated druggable and difficult targets in the second. We
              further identified novel druggable transcription factors with
              implications for cancer therapy. DrugFEATURE provides useful
              insight for drug discovery, by evaluating druggability and
              suggesting specific regions for interacting with drug-like
              molecules.CPT: Pharmacometrics Systems Pharmacology (2014) 3, e93;
              doi:10.1038/psp.2013.66; published online 22 January 2014. 
            </p>
          </abstract_text>
        </abstract>
      </abstracts>
    </fullrecord_metadata>
    <item xmlns:xsi='http://www.w3.org/2001/XMLSchema-instance' Owner='NLM' Status='PubMed-not-MEDLINE' coll_id='MEDLINE' xsi:type='itemType_medline'>
      <MedlineJournalInfo>
        <Country>United States</Country>
        <NlmUniqueID>101580011</NlmUniqueID>
        <ISSNLinking>2163-8306</ISSNLinking>
      </MedlineJournalInfo>
      <DateCreated>2014-01-23</DateCreated>
      <DateCompleted>2014-01-23</DateCompleted>
      <DateRevised>2016-10-19</DateRevised>
      <Affiliation>Department of Genetics, Stanford University, Stanford, California, USA.</Affiliation>
      <OtherID Source='NLM'>PMC3910014</OtherID>
    </item>
  </static_data>
  <dynamic_data>
    <citation_related>
      <tc_list>
        <silo_tc coll_id='MEDLINE' local_count='0'/>
      </tc_list>
    </citation_related>
    <cluster_related>
      <identifiers>
        <identifier type='doi' value='10.1038/psp.2013.66'/>
        <identifier type='pmid' value='MEDLINE:24452614'/>
      </identifiers>
    </cluster_related>
  </dynamic_data>
</REC>
```

File: lib/tasks/wos.rake [hunk 1]
```diff
@@ -0,0 +1,43 @@
+namespace :wos do
+  # @return [Clarivate::LinksClient]
+  def links_client
```
Comment (atz) [inline lib/tasks/wos.rake]: defining methods in rake files is an anti-pattern

Comment (darrenleeweber) [reply lib/tasks/wos.rake]: So what do you recommend as the pro-pattern for these simple util methods that make console life a happy dance?

Comment (darrenleeweber) [reply lib/tasks/wos.rake]: Whatever you do recommend, I hope your open to working on changing all the things too, e.g.
```
$ git grep 'def' lib/tasks/
lib/tasks/cap_cutover.rake:  def find_issn_id_identifiers(identifiers_array)
lib/tasks/cap_cutover.rake:  def build_pub_from_sw_and_pubmed(pub)
lib/tasks/cap_cutover.rake:  def build_pub_from_missing_sw_and_pubmed(pub)
lib/tasks/cap_cutover.rake:  def self.convert_manual_publication_row_to_hash(cap_pub_data_for_this_pub, author_id)
lib/tasks/pubmed.rake:  def client
lib/tasks/sul.rake:    method = args[:method] || "rebuild_pub_hash" # default to rebuilding pub_hash, could also rebuild_authorship
lib/tasks/sul.rake:    raise "Method #{method} not defined" unless Publication.new.respond_to? method
lib/tasks/sw.rake:  def harvester
lib/tasks/sw.rake:    args.with_defaults(starting_author_id: 1, ending_author_id: -1)
lib/tasks/sw.rake:    institution = sciencewire_harvester.default_institution
```

Comment (atz) [reply lib/tasks/wos.rake]: Yes, I already have work removing 4 of those.

Comment (darrenleeweber) [reply lib/tasks/wos.rake]: Done.

**Comment by darrenleeweber** [comment]
So, yes, methods in namespaces for rake tasks can leak.  For example, while coding this I had a typo in
```ruby
WebOfScience::Queries.new(client)
```
There was no `client` defined in this `wos` namespace.  But, it exists somewhere, because the `wos_queries` instance was initialized OK it got a `PubmedClient` instance.

**Pull request merged**